db/database_stats.go

```diff
@@ -0,0 +1,152 @@
+package db
+
+import (
+	"expvar"
+
+	"github.com/couchbase/sync_gateway/base"
+)
+
+// References into expvar stats that correspond to a particular database context.  To make it easier to
+// get references to the expvar maps that compose the database stats, keep explicit references to all
+// contained expvar maps.  When the entire structure needs to be added to a parent expvar map, convert
+// to an expvar map by calling the ExparMap() method.
+type DatabaseStats struct {
+	StatsCache                *expvar.Map
+	StatsDatabase             *expvar.Map
+	StatsDeltaSync            *expvar.Map
+	StatsConvergenceImport    *expvar.Map
+	StatsCblReplicationPush   *expvar.Map
+	StatsCblReplicationPull   *expvar.Map
+	StatsCblReplicationCommon *expvar.Map
+	StatsSecurity             *expvar.Map
+	StatsGsiViews             *expvar.Map
+}
+
```

Comment:
Here's a proposed change for lazy initialization:

https://github.com/couchbase/sync_gateway/pull/3837/commits/75abf169208ae925047fa0ebc93fb81c0ae42f00

If a stats group under the db stats is never accessed, it will remain empty.  For example, in startup it currently is:

```
  "syncgateway": {
    "global": {
      "resource_utilization": {...}
    
    },
    "per_db": {
      "db": {
        "cache": {
          "chan_cache_perf": 0,
          "chan_cache_utilization": 0,
          "num_skipped_seqs": 0,
          "rev_cache_hits": 0,
          "rev_cache_misses": 0,
          "rev_cache_utilization": 0
        }
      },
``` 

the reason `cache` is added by default is due to this access:

```
	context.revisionCache = NewRevisionCache(
		options.RevisionCacheCapacity,
		context.revCacheLoader,
		context.DbStats.StatsCache(),
	)
```

This could be changed by passing in the `context.DbStats` struct and delaying the call to `StatsCache()` until any revision cache stats are actually recorded, but the tradeoff is that it's nice to constrain the `RevisionCache` to only have access to the `StatsCache()` rather than the full set of database stats.

Once a group is accessed for the first time, all fields in the group will be initialized to their zero values (whatever makes sense for that stat)

In the case you mentioned, "push replication for a read-only deployment", it should remain empty indefinitely since that stats group will never be accessed.